### DISCUSSION TITLE
Multicast

### DISCUSSION THREAD

**Discussion by mdavies2310** [OpenDDS/OpenDDS#4222] [Q&A] [answered]
Hi,
I have a publisher that uses multicast(I do not have access to the code) and I am creating a subscriber to read/listen for data from the publisher. Does the domain ID need to match the publisher? Or does that not matter? 

I created a test publisher and I can only receive data when the IDs match but I thought it shouldn't matter with multicast?

My rtps.ini file looks like this. for both the pub and sub.

```ini
[common]
DCPSDefaultDiscovery=DEFAULT_RTPS
DCPSGlobalTransportConfig=$file
[transport/the_rtps_transport]
transport_type=rtps_udp
use_multicast=1
```

**Comment by mitza-oci** [comment, marked as answer]
Domain IDs must match regardless of transport.  It's part of the overall DDS spec semantics for what it means to be a participant in a domain.

**Reply by mdavies2310** [reply]
Ok thank you

**Reply by mdavies2310** [reply]
So then multicast would mean many participants can use that same domain id?

**Reply by jrw972** [reply]
If you want participants to communicate, then they need to use the same domain id.  If you are using RTPS discovery, the domain ID influences discovery, i.e., if you use different domain ids, they will not discover each other.
